### PR TITLE
Milestone3 - Build an in-memory RESTful API

### PR DESCRIPTION
**Pull request by javierLiarte**
- Created mocked in memory Event repository
- Created event router using in-memory repository functions
- Refactored to use Promises in event repository and consume them in event router
- Refactored error generation in event repository to use error handling middleware in event router

### CODE REVIEW

File: event_repository.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+const event = require('./event')
+
+const ev1 = new event('ev1', 'This is 1st event', 'Short event 1 desciption', Date.now())
+const ev2 = new event('ev2', 'This is 2nd event', 'Description of 2nd event', Date.now())
+const ev3 = new event('ev3', 'This is last event', 'Last event for now', Date.now())
+
+let events = [ev1, ev2, ev3]
```
Comment (gootyfer) [inline event_repository.js]: You can use `const` here since the variable is never re-assigned

File: event_repository.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    return new Promise((resolve, reject) => {
+        resolve(events);
+    });
+}
+
+function getEventById(id) {
+    return new Promise((resolve, reject) => {
+        const filteredEvent = events.filter((item) => {
+            return item.id === id;
+        })[0];
```
Comment (gootyfer) [inline event_repository.js]: In this case I prefer to use the array method `find`that is more suitable when you search for just one result

File: event_repository.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
+const event = require('./event')
```
Comment (gootyfer) [inline event_repository.js]: I'd use capital first `Event`

**Review by gootyfer — COMMENTED** [review]
Nice use of promises.
Overall an excellent work!

File: event_repository.js (hunk 1 above)
Comment (Jesusz0r) [reply event_repository.js]: We re-asigned a new value to the event variable in line 60!

File: event_repository.js [hunk 4]
```diff
@@ -14,7 +14,7 @@ function getAllEvents() {
 
 function getEventById(id) {
     return new Promise((resolve, reject) => {
-        const filteredEvent = events.filter((item) => {
+        const filteredEvent = events.find((item) => {
             return item.id === id;
         })[0];
```
Comment (javierLiarte) [inline event_repository.js]: According to MDN, find "returns the value of the first element in the array that satisfies the provided testing function", so we must get rid of [0] here.

I will change this and merge the PR to master :)

**Comment by javierLiarte** [comment]
Thanks for your comments @gootyfer :)

**Pull request merged**